### PR TITLE
feature/523 editnew

### PR DESCRIPTION
**Pull request by gentee**

### CODE REVIEW

File: packages/smart/smart_p.go [hunk 1]
```diff
@@ -614,19 +614,19 @@ func Substr(s string, off int64, slen int64) string {
 
 // Activate sets Active status of the contract in smartVM
 func Activate(sc *SmartContract, tblid int64, state int64) error {
-	if sc.TxContract.Name != `@1ActivateContract` {
-		log.WithFields(log.Fields{"type": consts.IncorrectCallingContract}).Error("ActivateContract can be only called from @1ActivateContract")
-		return fmt.Errorf(`ActivateContract can be only called from @1ActivateContract`)
+	if !accessContracts(sc, "ActivateContract", "DeactivateContract") {
```
Comment (ghost) [inline packages/smart/smart_p.go]: strings have to be moved to constants

File: packages/smart/smart_p.go [hunk 2]
```diff
@@ -614,19 +614,19 @@ func Substr(s string, off int64, slen int64) string {
...
 	ActivateContract(tblid, state, true)
 	return nil
 }
 
 // DeactivateContract sets Active status of the contract in smartVM
 func Deactivate(sc *SmartContract, tblid int64, state int64) error {
-	if sc.TxContract.Name != `@1DeactivateContract` {
-		log.WithFields(log.Fields{"type": consts.IncorrectCallingContract}).Error("DeactivateContract can be only called from @1DeactivateContract")
-		return fmt.Errorf(`DeactivateContract can be only called from @1DeactivateContract`)
+	if !accessContracts(sc, "ActivateContract", "DeactivateContract") {
```
Comment (ghost) [inline packages/smart/smart_p.go]: the same

File: packages/smart/smart_p.go [hunk 3]
```diff
@@ -692,6 +692,23 @@ func JSONToMap(input string) (map[string]interface{}, error) {
 	return ret, nil
 }
 
+func RollbackContract(sc *SmartContract, name string) error {
+	if !accessContracts(sc, "NewContract", "Import") {
```
Comment (ghost) [inline packages/smart/smart_p.go]: one more plain strings

**Review by ghost — CHANGES_REQUESTED** [review]

File: packages/smart/smart_p.go (hunk 1 above)
Comment (gentee) [reply packages/smart/smart_p.go]: Fixed, contract names moved to const.

**Pull request merged**